include/ifc/reader.hxx

```diff
@@ -29,7 +29,7 @@ namespace ifc {
         {
             const auto byte_offset = ifc::to_underlying(offset);
             const auto& contents   = ifc.contents();
-            IFCASSERT(byte_offset < contents.size());
+            IFCASSERT(byte_offset <= contents.size() && (contents.size() - byte_offset) >= sizeof(T));
 
```

Comment:
1. The change to `byte_offset <= contents.size()` will allow buffer overrun.
2.  How do you know the new assertion fixes the original bug report?

Reply:
You strengthened the assertion, which is good, but that is not the root cause of the failure.  The assertion is only signaling a symptom, a sign that an error occurred much earlier.  Keep the assertion, and keep looking for the root cause of the bug.